src/main/resources/liquibase/crop_changelog/studies_TBD.xml

```diff
@@ -0,0 +1,626 @@
+        </preConditions>
+        <comment>Migrate ENTRY_CODE from listdata to list_data_details table as entry details variable</comment>
+        <sql dbms="mysql" splitStatements="true">
+             <!-- Migrate ENTRY_CODE from listdata to list_data_details -->
+			INSERT INTO list_data_details(variable_id, lrecid, value, created_by)
+            SELECT 8300, r.lrecid, r.entrycd, ln.listuid FROM listdata r inner join listnms ln on ln.listid = r.listid;
+
+			<!-- Insert ENTRY_CODE variable to list_data_view -->
+			INSERT INTO list_data_view(listid,cvterm_id,type_id)
+			SELECT distinct(ld.listid), 8300, 1815 FROM listnms ld;
+
+			<!-- Remove ENTRY_CODE as default view -->
+			DELETE FROM list_data_default_view WHERE name = 'ENTRY_CODE';
+
+			<!-- Remove ENTRY_CODE as static column -->
+			DELETE FROM list_data_view WHERE static_id = 8300;
+        </sql>
+  </changeSet>
+
+	<changeSet author="lkovacic" id="studies-TBD-4">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+				SELECT COUNT(1) > 0
+				FROM project p
+						 INNER JOIN projectprop pp ON p.project_id = pp.project_id AND p.dataset_type_id = 4
+						 INNER JOIN stock s ON p.parent_project_id = s.project_id
+						 INNER JOIN stockprop sp ON s.stock_id = sp.stock_id
+				WHERE pp.variable_id = 8300 AND sp.type_id = 8300;
+            </sqlCheck>
+		</preConditions>
+		<comment>Entry_code migration as stock property</comment>
+		<sql>
+			<!-- Only migrate the entry_code that the user added in the study -->
+			INSERT INTO stockprop(stock_id, type_id, value)
+				SELECT stock_id, 8300, s.value FROM project p
+						 INNER JOIN projectprop pp ON p.project_id = pp.project_id AND p.dataset_type_id = 4
+						 INNER JOIN stock s ON p.parent_project_id = s.project_id
+				WHERE pp.variable_id = 8300;
+		</sql>
+	</changeSet>
+
+    <changeSet author="lkovacic" id="studies-TBD-5">
+        <preConditions onFail="MARK_RAN">
+			<columnExists tableName="stock" columnName="value"/>
+		</preConditions>
+		<comment>Rename column 'value' to 'value_bkp' in table 'stock'</comment>
+		<sql>
+			ALTER TABLE stock CHANGE value value_bkp varchar(255) null;
+		</sql>
+	</changeSet>
+
+    <changeSet author="lkovacic" id="studies-TBD-6">
+        <preConditions onFail="MARK_RAN">
+			<columnExists tableName="stock" columnName="type_id"/>
+		</preConditions>
+		<comment>Rename column 'type_id' to 'type_id_bkp' in table 'stock' and make it nullable</comment>
+		<sql>
+			ALTER TABLE stock CHANGE type_id type_id_bkp int null;
+		</sql>
+	</changeSet>
+
+	<changeSet author="lkovacic" id="studies-TBD-7">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT COUNT(*)
+				FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS
+				WHERE TABLE_NAME = 'stock'
+				  AND CONSTRAINT_NAME = 'stock_idx1'
+				  AND TABLE_SCHEMA = DATABASE();
+			</sqlCheck>
+		</preConditions>
+		<comment>Drop unique index 'stock_idx1' in 'stock' table</comment>
```

Comment:
@lik-83 I guess we should also drop stock_idx5, because it refers to type_id only